lib/kafka/consumer_interceptors.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Kafka
+  # Holds a list of Kafka::ConsumerInterceptor
+  # and wraps calls to a chain of custom consumer interceptors.
+  class ConsumerInterceptors
+    def initialize(interceptors:, logger:)
+      @interceptors = interceptors
+      @logger = TaggedLogger.new(logger)
+    end
+
+    # This is called from Kafka::Consumer once the batches are fetched.
+    #
+    # @param batches [Array<Kafka::FetchedMessage>]
+    # @return [Array<Kafka::FetchedMessage>] intercepted batches.
+    def on_consume(batches:)
+      intercepted_batches = batches
```

Comment:
You could just name the method argument `intercepted_batches` and remove that line